       '''
       Custom exceptions raised by pytz.
    1: '''
       
       __all__ = [
    1:     'UnknownTimeZoneError', 'InvalidTimeError', 'AmbiguousTimeError',
    1:     'NonExistentTimeError',
       ]
       
       
    2: class UnknownTimeZoneError(KeyError):
           '''Exception raised when pytz is passed an unknown timezone.
       
           >>> isinstance(UnknownTimeZoneError(), LookupError)
           True
       
           This class is actually a subclass of KeyError to provide backwards
           compatibility with code relying on the undocumented behavior of earlier
           pytz releases.
       
           >>> isinstance(UnknownTimeZoneError(), KeyError)
           True
    1:     '''
    1:     pass
       
       
    2: class InvalidTimeError(Exception):
    1:     '''Base class for invalid time exceptions.'''
       
       
    2: class AmbiguousTimeError(InvalidTimeError):
           '''Exception raised when attempting to create an ambiguous wallclock time.
       
           At the end of a DST transition period, a particular wallclock time will
           occur twice (once before the clocks are set back, once after). Both
           possibilities may be correct, unless further information is supplied.
       
           See DstTzInfo.normalize() for more info
    1:     '''
       
       
    2: class NonExistentTimeError(InvalidTimeError):
           '''Exception raised when attempting to create a wallclock time that
           cannot exist.
       
           At the start of a DST transition period, the wallclock time jumps forward.
           The instants jumped over never occur.
    1:     '''
